sim/battle.ts

```diff
@@ -2342,7 +2342,7 @@ export class Battle extends Dex.ModdedDex {
 				}
 			} else if (action.choice === 'switch' || action.choice === 'instaswitch') {
 				if (typeof action.pokemon.switchFlag === 'string') {
-					action.sourceEffect = this.getEffect(action.pokemon.switchFlag);
+					action.sourceEffect = (this.getMove(action.pokemon.switchFlag as ID) as unknown) as PureEffect;
```

Comment:
We get around this is `Dex.getEffect` with `// @ts-ignore`, but apparently thats really bad and `Foo -> unknown -> Bar` is the 'correct' way to abuse the type system.

Reply:
I wouldn't use parens for the double cast. What does "apparently" imply here?

Reply:
Ideally, with the right other changes, this can be:

```suggestion
					action.sourceEffect = this.getMove(action.pokemon.switchFlag!) as any;
```

Reply:
Not sure why you'd want to cast to `any`.

Reply:
It's an easier cast and approximately as safe.

Reply:
Done.